### PR TITLE
Fix random encoder lockups

### PR DESCRIPTION
**Pull request by ZwX1616**
Ticis seem to be happy finally. Output files on cici also have exactly the same hash as old ones.

### CODE REVIEW

File: selfdrive/loggerd/encoder.c [hunk 1]
```diff
@@ -610,9 +611,8 @@ void encoder_close(EncoderState *s) {
 
       OMX_BUFFERHEADERTYPE* in_buf = queue_pop(&s->free_in);
       in_buf->nFilledLen = 0;
-      in_buf->nOffset = 0;
```
Comment (geohot) [inline selfdrive/loggerd/encoder.c:613]: Hmm, is this uninitted now?

Comment (ZwX1616) [reply selfdrive/loggerd/encoder.c:613]: Oh it is initted when first called by encoder_encode_frame. If a segment is somehow shorter than the amount of buffers then pBuffer will be nullptr anyways. Shouldn't matter either way?

Comment (geohot) [reply selfdrive/loggerd/encoder.c:613]: Unless you have to remove, I'd keep setting it. That timestamp thing seems very plausibly like the issue, I remember something like this when I did the qcameras.

Comment (ZwX1616) [reply selfdrive/loggerd/encoder.c:613]: Sounds good. I will just add it back.
Added some more changes (replaced a hack with something more reasonable). It seems qcom/qcom2 don't work in the exact same way :|

**Pull request merged**